keadm/cmd/keadm/app/cmd/ctl/client/pod.go

```diff
@@ -30,42 +31,36 @@ type PodRequest struct {
 }
 
 func (podRequest *PodRequest) GetPod() (*corev1.Pod, error) {
-	request := Request{
-		Method: "GET",
-		Path: "/" + CoreAPIPrefix + "/" + CoreAPIGroupVersion.Version +
-			"/namespaces/" + podRequest.Namespace + "/pods/" + podRequest.PodName,
+	kubeClient, err := KubeClient()
+	if err != nil {
+		return nil, err
 	}
 
-	pod, err := request.ResponseToPod()
+	pod, err := kubeClient.CoreV1().Pods(podRequest.Namespace).Get(context.Background(), podRequest.PodName, metaV1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
 	return pod, nil
 }
 
 func (podRequest *PodRequest) GetPods() (*corev1.PodList, error) {
-	var request Request
+	kubeClient, err := KubeClient()
+	if err != nil {
+		return nil, err
+	}
 	if podRequest.AllNamespaces {
-		request = Request{
-			Method: "GET",
-			Path:   "/" + CoreAPIPrefix + "/" + CoreAPIGroupVersion.Version + "/pods",
-		}
-	} else {
-		request = Request{
-			Method: "GET",
-			Path: "/" + CoreAPIPrefix + "/" + CoreAPIGroupVersion.Version +
-				"/namespaces/" + podRequest.Namespace + "/pods",
+		podList, err := kubeClient.CoreV1().Pods(metaV1.NamespaceAll).List(context.TODO(), metaV1.ListOptions{
```

Comment:
Is it better if the variable context is inputed by the caller?

Reply:
ok